core/src/main/java/org/infinispan/persistence/spi/NonBlockingStore.java

```diff
@@ -32,33 +79,52 @@
 @Experimental
 public interface NonBlockingStore<K, V> {
 
+   /**
+    * Enumeration defining the various characteristics of the underlying store to communicate what features it may
+    * or may not support.
+    */
    enum Characteristic {
       /**
        * Whether this cache can be shared between multiple nodes. An example would be an external system, such as
-       * a database.
+       * a database. This characteristic is here solely for validation of the store configuration.
        */
       SHAREABLE,
       /**
        * If this store only supports being read from.  Any write based operations will never be invoked on this store.
+       * No optional methods map to this characteristic. The {@link #write(int, MarshallableEntry)},
+       * {@link #delete(int, Object)}, {@link #bulkWrite(int, Publisher)}, and {@link #bulkDelete(int, Publisher)} methods
+       * will not be invoked on a store that has this characteristic.
+       * <p>
+       * If this characteristic is provided, {@link #TRANSACTIONAL} will be ignored if also present.
        */
       READ_ONLY,
       /**
        * If this store only supports being written to. Any read based operations will never be invoked on this store.
+       * No optional methods map to this characteristic. The {@link #load(int, Object)} and
+       * {@link #containsKey(int, Object)} methods will not be invoked on a store that has this characteristic.
+       * <p>
+       * If this characteristic is provided, {@link #BULK_READ} will be ignored if also present.
```

Comment:
I think we should describe characteristics being ignored one way, i.e. describe on a given characteristic if another can override it. So in this case just add the Note to BULK_READ that it is ignored if WRITE_ONLY is present, no need for both Javadocs to contain the note.

Reply:
Okay. I was trying to just be overly cautious as to how they were affected, I wondered about doing this.